### PR TITLE
Dependency: Install pymatgen-analysis-defect for defect-related functionality

### PR DESCRIPTION
**Pull request by HuangJiameng**
- add `pymatgen-analysis-defect` and `pybind11` in `setup.py`.
- add reminder for user to install `pymatgen-analysis-defect` if this package is not in their environment.

Note: If `pymatgen-analysis-defect` is installed without `pybind11`, pip will raise errors.

### CODE REVIEW

**Comment by wanghan-iapcm** [comment]
In the test the verions of pymatgen is hard coded. Please update accordingly.
https://github.com/deepmodeling/dpgen/blob/8dea29ef125f66be9641afe5ac4970433a9c9ce1/.github/workflows/test.yml#L13

File: tests/auto_test/test_interstitial.py [hunk 1]
```diff
@@ -7,7 +7,13 @@
...
 from pymatgen.io.vasp import Incar
 from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
-from pymatgen.analysis.defects.core import Interstitial as pmg_Interstitial
+try: 
+    from pymatgen.analysis.defects.core import Interstitial as pmg_Interstitial
+except ModuleNotFoundError:
+    print("pymatgen==2022.7.19 is the final release with `pymatgen.analysis.defects` module. Please check the version of pymatgen.")
+    print("Please install `pymatgen-analysis-defects`.")
+    print("Kindly reminder: `pybind11>=2.4` need to be installed previously.")
+    os._exit(0)
```
Comment (njzjz) [inline tests/auto_test/test_interstitial.py:16]: Use raise instead

File: setup.py [hunk 2]
```diff
@@ -29,6 +29,8 @@
     'dpdispatcher>=0.3.11',
     'netCDF4',
     'dargs>=0.2.9',
+    'pybind11>=2.4',
```
Comment (njzjz) [inline setup.py:32]: Why do we need it?

Comment (HuangJiameng) [reply setup.py:32]: Installing `pymatgen-analysis-defects` without `pybind11` will cause errors.

**Comment by HuangJiameng** [comment]
Hmm... Some problems with GROMACS are raised. How can I fix it?

**Comment by njzjz** [comment]
> Hmm... Some problems with GROMACS are raised. How can I fix it?

I don't think so.. It looks that some autotest modules are not compatible with the latest pymatgen.

**Comment by njzjz** [comment]
I think we should fix it. There's two places that need attention.

**Review by kevinwenminion — APPROVED** [review]

**Comment by HuangJiameng** [comment]
Could you fix the autotest modules to make it compatible with the latest pymatgen? Thanks. @kevinwenminion

**Comment by ZLI-afk** [comment]
test

**Comment by HuangJiameng** [comment]
Since fixes are made in #963, this pr should be closed.

**Pull request closed without merging**